### DISCUSSION TITLE
isDisplayed() is return false

### DISCUSSION THREAD

**Discussion by neerajkumarlad** [webdriverio/webdriverio#8594] [Community]
Hi 

I am facing wierd issue. I am not getting this is becouse my code or something else.
Issue is, When i pass element in my custom isElementDisplayed(elem) to check element is displayed or not, but i am getting always false even element displayed. 

Here is code,

```
  async selectMoolyaCompany() {
    let isCompanyDisplayed = await Actions.isElementDisplayed(this.selectCompany);
    console.log("direct displaye : ", await this.selectCompany.isDisplayed());
    console.log("isCompanyDisplayed:", isCompanyDisplayed);
    await Actions.waitForElementForDisplay(this.selectCompany, 50000);
    await Actions.doClickOn(this.selectCompany);
  }
```

**_Here is my custom code:_**

  ```
async isElementDisplayed(elem) {
    let isDisplayed = await elem.isDisplayed();
    console.log("Element isDisplayed Value  ==>  ", isDisplayed);
    return isDisplayed;
  }

  async doClickOn(elem) {
    await this.isElementDisplayed(elem);
    console.log("Clicking On ==> ", await elem.getText());
    await elem.click();
  }

  async waitForElementForDisplay(elem, elementWaitTimeout) {
    await elem.waitForDisplayed({ timeout: elementWaitTimeout });
  }
```

Execution Log: 
Line 1 -- [0-0] Element isDisplayed Value  ==>   false
Line 2 -- **[0-0] direct displaye :  false
Line 3 -- [0-0] isCompanyDisplayed: false**
Line 4 -- **[0-0] Element isDisplayed Value  ==>   true**
Line 5 -- [0-0] Clicking On ==>  Moolya Tests

If you observed, 
First log line, It says false => however, element is displayed
Second log line, I am calling direct isDisplayed() method == > It says false => however, element is displayed
Third log line, Called custom method isElementDisplayed(). == > It says false => however, element is displayed
Fourth log line, I used waitForElementForDisplay() where it displayed true
Fifth log line, It clicked on element

Whole selectMoolyaCompany() get executed on same page. 

I am not sure why its happening. what's my mistake.

**Comment by erwinheitzman** [comment]
There could be many things like for example it could be overlapped by another (almost transparent) element.

Waiting for the element to be fully displayed therefor might work as it will wait until all conditions are met.

You could execute the `isDisplayed` command after waiting for it to be displayed and in that case it should return true.
